core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java

```diff
@@ -1060,6 +1061,25 @@ public void workerHeartbeatTest() throws Exception {
   }
 
   /**
+   * Tests the {@link FileSystemMaster#getFilesPersistInProgress()} method.
+   */
+  @Test
+  public void getFilesPersistInProgressTest() throws Exception {
+    createFileWithSingleBlock(ROOT_FILE_URI);
+
+    long fileId = mFileSystemMaster.getFileId(ROOT_FILE_URI);
+    mFileSystemMaster.scheduleAsyncPersistence(ROOT_FILE_URI);
+    Assert.assertTrue(mFileSystemMaster.getFilesPersistInProgress().isEmpty());
+
+    mFileSystemMaster.workerHeartbeat(mWorkerId1, Lists.<Long>newArrayList());
```

Comment:
Use `HeartbeatScheduler` to schedule heartbeat, your current approach may cause flaky test results due to uncertainty of the start and end of heartbeat. See `lostFilesDetectionTest` for an example usage.